flink-end-to-end-tests/flink-streaming-kafka-test-base/src/main/java/org/apache/flink/streaming/kafka/test/base/KafkaExampleUtil.java

```diff
@@ -45,6 +45,7 @@ public static StreamExecutionEnvironment prepareExecutionEnv(ParameterTool param
 		env.getConfig().disableSysoutLogging();
 		env.getConfig().setRestartStrategy(RestartStrategies.fixedDelayRestart(4, 10000));
 		env.enableCheckpointing(5000); // create a checkpoint every 5 seconds
+		env.getCheckpointConfig().setTolerableCheckpointFailureNumber(Integer.MAX_VALUE - 1);
```

Comment:
This can be simple `Integer.MAX_VALUE` if you change the precondition for max failures to `<=`